public/app/features/alerting/unified/RedirectToRuleViewer.test.tsx

```diff
@@ -0,0 +1,143 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import { Provider } from 'react-redux';
+import { Router } from 'react-router-dom';
+import { DataSourceJsonData, PluginMeta } from '@grafana/data';
+import { locationService } from '@grafana/runtime';
+import { RedirectToRuleViewer } from './RedirectToRuleViewer';
+import { configureStore } from 'app/store/configureStore';
+import { typeAsJestMock } from '../../../../test/helpers/typeAsJestMock';
+import { useCombinedRulesMatching } from './hooks/useCombinedRule';
+import { CombinedRule, Rule } from '../../../types/unified-alerting';
+import { PromRuleType } from '../../../types/unified-alerting-dto';
+import { getRulesSourceByName } from './utils/datasource';
+
+jest.mock('./hooks/useCombinedRule');
+jest.mock('./utils/datasource');
+jest.mock('react-router-dom', () => ({
+  ...(jest.requireActual('react-router-dom') as any),
+  Redirect: jest.fn(({}) => `Redirected`),
+}));
+
+const store = configureStore();
+const renderRedirectToRuleViewer = () => {
+  return render(
+    <Provider store={store}>
+      <Router history={locationService.getHistory()}>
+        <RedirectToRuleViewer {...mockRoute('prom alert', 'test prom')} />
+      </Router>
+    </Provider>
+  );
+};
+
+const mockRuleSourceByName = () => {
+  typeAsJestMock(getRulesSourceByName).mockReturnValue({
+    name: 'prom test',
+    type: 'prometheus',
+    uid: 'asdf23',
+    id: 1,
+    meta: {} as PluginMeta,
+    jsonData: {} as DataSourceJsonData,
+  });
+};
+
+describe('Redirect to Rule viewer', () => {
+  it('should list rules that match the same name', () => {
+    typeAsJestMock(useCombinedRulesMatching).mockReturnValue({
+      result: mockedRules,
+      loading: false,
+      dispatched: true,
+      requestId: 'A',
+      error: undefined,
+    });
+    mockRuleSourceByName();
+    renderRedirectToRuleViewer();
+    expect(screen.getByText('Alerting / Find rule')).toBeInTheDocument();
```

Comment:
this is just the title? What about checking what rule cards are rendered?

Reply:
Uh, yeah that makes sense